packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -71,10 +75,22 @@ export async function forceSourceDeploySourcePath(sourceUri: vscode.Uri) {
       return;
     }
   }
+
+  const toolingDeployEnabled = sfdxCoreSettings.getToolingDeploys();
+  // this supported types logic is temporary until we have a way of generating the metadata type from the path
+  // once we have the metadata type we can check to see if it is a toolingsupportedtype from that util
+  const supportedType =
+    path.extname(sourceUri.fsPath) === APEX_CLASS_EXTENSION ||
+    sourceUri.fsPath.includes(`${APEX_CLASS_EXTENSION}-meta.xml`);
+  const executor =
+    toolingDeployEnabled && supportedType
+      ? new DeployRetrieveExecutor()
```

Comment:
I'm not sure we need an entirely new CommandletExecutor for this. Can we perform the tooling deploy logic in the `ForceSourceDeploySouthPathExecutor`? You might be able to just define the execute() method in `ForceSourceDeploySourcePathExecutor` where you check to see if the type is supported for tooling, and if it is run the tooling logic. Otherwise call `super.execute()` to do the fallback CLI way.

Reply:
Ah I see what you mean, let me see how I can consolidate